### PR TITLE
add flagd tutorial

### PR DESCRIPTION
**Pull request by agardnerIT**
## This PR
- Adds a demo for flagd

### Notes

[Try it here...](https://killercoda.com/agardnerit/scenario/openfeature-custom-backend-flagd)

Temporarily (so you can see what this tutorial does), I've hosted it on my account (see above).

If this PR is merged, I will remove my copy in favour of this one.

### CODE REVIEW

File: flagd-demo/assets/scripts/intro_foreground.sh [hunk 1]
```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+
+DEBUG_VERSION=1
+GITEA_VERSION=1.19
+TEA_CLI_VERSION=0.9.2
+FLAGD_VERSION=0.3.7
```
Comment (beeme1mr) [inline flagd-demo/assets/scripts/intro_foreground.sh:6]: Version 0.4.0 is available now.

File: flagd-demo/assets/scripts/intro_foreground.sh [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+   --user=openfeature \
+   --password=openfeature \
+   --url=http://0.0.0.0:3000 \
+   --token=$ACCESS_TOKEN > /dev/null 2>&1
+
+# Create an empty repo called 'flags'
+# Clone the template repo
+cd ~
+tea repo create --name=flags --init=false
+git clone https://github.com/agardnerIT/OpenFeatureFlagsTemplate
```
Comment (beeme1mr) [inline flagd-demo/assets/scripts/intro_foreground.sh:149]: Can this be changed to an OpenFeature repo?

File: flagd-demo/intro.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# OpenFeature and flagd Demonstration
+
+This demonstration will show a GitOps-based system and approach for storing, changing and evaluating feature flags.
+
+The demo will use [flagd](https://github.com/open-feature/flagd), an OpenFeature compliant feature flag daemon & API layer but these principles could be used with any open source or commercial feature flag vendor.
```
Comment (beeme1mr) [inline flagd-demo/intro.md:5]:
```suggestion
The demo will use [flagd](https://github.com/open-feature/flagd), an OpenFeature compliant feature flag daemon & API layer but these principles apply to many open source or commercial feature flag vendors.
```

File: flagd-demo/step1.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+## Your Username and Password
+Certain commands will prompt for a username and password.
```
Comment (beeme1mr) [inline flagd-demo/step1.md:2]: Is it possible to set this automatically in the gift config so the user didn't have to worry about it?

File: flagd-demo/step2.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+flagd is compatible with `gRPC`{{}} and `HTTP`{{}} and has native support for metrics using Prometheus. In this demo we will read flags directly from a Git repo using `HTTPS`{{}}.
+
+## Start flagd
+
+Start flagd and ask it to read files from a JSON source hosted online:
+
+```
+flagd start \
+  --port 8013 \
```
Comment (beeme1mr) [inline flagd-demo/step2.md:13]: You don't need this since it's the default value but it doesn't hurt anything to include it.

**Review by beeme1mr — APPROVED** [review]
It looks great. I just left some minor feedback. It would be awesome if you could record a video of you walking through this tutorial. We could embed it here and in the flagd docs.

**Comment by agardnerIT** [comment]
Changes made but please do not merge until we have an outcome on https://github.com/open-feature/flagd/pull/467

If the above linked PR is merged, I can use an OpenFeature-based repo for the code.
If it isn't I'll need to either create the example flags on the fly or use my own repo. I'd rather use OpenFeature rather than creating many copies of config so hopefully we can all agree and merge 467.

**Comment by agardnerIT** [comment]
This is ready for review. The final version:

- Uses flagd v4.4.4
- Adds a tutorial step which demonstrates the use of flagd pulling from multiple backends
- Adds links in the final step to the k8s operator killercoda tutorial + OFO docs
- Adds links to the flagd language specific providers
- Adds links to the community
- Adds Golang code sample in tutorial as a teaser for how to get started for real

When Pete's "5 minutes to feature flagging" tutorial is ready (#8) I'll add links to that too as it's a natural fit. That will happen as a different PR.

**Pull request merged**